app/views/sessions/purchase_cart.html.erb

```diff
@@ -7,7 +7,11 @@
     </tr>
     <% if session[:cart] %>
         <% session[:cart].each do |item_id, quantity| %>
-        <% item = Item.find_by(id: item_id) %>
+            <% item = Item.find_by(id: item_id) %>
+            <!-- Line below generates receipt for purchasing this item, if possible -->
+            <% if logged_in? %>
```

Comment:
@kevinchen93 That's a good point. I've improved on the above, but I'll think further on this.

There is quite a bit of logic in the view, but that's primarily because the view is associated with "triggering" the logic. I.e., the logic of emptying the cart and e.t.c.
It may make sense to have that associated with the button press on the previous screen though.
Will think about this further.

In terms of the receipt, we've currently set it up so that non-logged in users can view their receipt using a unique 10 digit code. This 10 digit code has a possible space of 64 characters.